sdk/core/core-http/lib/credentials/sasServiceClientCredentials.ts

```diff
@@ -0,0 +1,65 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+
+import { HttpHeaders } from "../httpHeaders";
+import { WebResource } from "../webResource";
+import { ServiceClientCredentials } from "./serviceClientCredentials";
+import { Constants } from "../util/constants";
+import { generateKey } from "../util/crypto";
+
+const HeaderConstants = Constants.HeaderConstants;
+
+export class SasServiceClientCredentials implements ServiceClientCredentials {
```

Comment:
@bterlson, 

I recall Scott saying that we do not want sas credentials to be defined from any of the core libraries and that each service was to have their sas implementation in their own libraries. If that is indeed true, then this should move out of core-http and into Service Bus library

Reply:
Updated this and rebased as well - can revert commit if we want to house the SAS stuff back in core